aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -997,6 +1047,49 @@ The following examples use the <xref:Microsoft.AspNetCore.Server.Kestrel.Core> n
 using Microsoft.AspNetCore.Server.Kestrel.Core;
 ```
 
+Kestrel options, which are configured in C# code in the following examples, can also be set using a [configuration provider](xref:fundamentals/configuration/index). For example, the File Configuration Provider can load Kestrel configuration from a file (*appsettings.json* or *appsettings.{Environment}.json*):
+
+```json
+{
+  "Kestrel": {
+    "Limits": {
+      "MaxConcurrentConnections": 100,
+      "MaxConcurrentUpgradedConnections": 100
+    },
+    "ShutdownTimeout": "00:00:20"
```

Comment:
ShutdownTimeout hasn't been a property on KestrelServerOptions since 2.0. Now we configure it via the WebHost: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-3.0#shutdown-timeout

There are no 3.0 API docs for KestrelServerOptions yet. [Here's the 2.2 doc](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserveroptions?view=aspnetcore-2.2), or you can just look at the [3.0 ref assembly source](https://github.com/aspnet/AspNetCore/blob/v3.0.0/src/Servers/Kestrel/Core/ref/Microsoft.AspNetCore.Server.Kestrel.Core.netcoreapp3.0.cs#L116) to see a more up-to-date list of properties.

If we don't want to show modification of AddServerHeader or AllowSynchronousIO, I guess setting DisableStringReuse to true is about the only thing left.